process/firehoseDataProcessor.go

```diff
@@ -89,25 +89,26 @@ func (dp *dataProcessor) saveBlock(marshalledData []byte) error {
 
 	log.Info("firehose: saving block", "nonce", header.GetNonce(), "hash", outportBlock.BlockData.HeaderHash)
 
-	_, err = fmt.Fprintf(dp.writer, "%s %s %d\n",
-		firehosePrefix,
-		beginBlockPrefix,
-		header.GetNonce(),
-	)
-	if err != nil {
-		return fmt.Errorf("could not write %s prefix , err: %w", beginBlockPrefix, err)
+	blockNum := header.GetNonce()
+	parentNum := blockNum - 1
+	if blockNum == 0 {
+		parentNum = 0
 	}
+	encodedMarshalledData := base64.StdEncoding.EncodeToString(marshalledData)
 
-	_, err = fmt.Fprintf(dp.writer, "%s %s %d %s %d %x\n",
+	_, err = fmt.Fprintf(dp.writer, "%s %s %d %s %d %s %d %d %s\n",
 		firehosePrefix,
-		endBlockPrefix,
-		header.GetNonce(),
+		blockPrefix,
+		blockNum,
+		hex.EncodeToString(outportBlock.BlockData.HeaderHash),
+		parentNum,
 		hex.EncodeToString(header.GetPrevHash()),
+		outportBlock.HighestFinalBlockNonce,
 		header.GetTimeStamp(),
-		marshalledData,
+		encodedMarshalledData,
 	)
 	if err != nil {
-		return fmt.Errorf("could not write %s prefix , err: %w", endBlockPrefix, err)
+		return fmt.Errorf("could not write %s , err: %w", blockPrefix, err)
```

Comment:
now that we have only entry line and there is just blockPrefix, i don't think it's useful anymore to have it into the error message

Reply:
right, will just propagate the error as it is